src/esp/bindings/CoreBindings.cpp

```diff
@@ -14,25 +14,177 @@ namespace esp {
+      .def(
+          "has_float",
+          [](Configuration& self, const std::string& key) {
+            return self.checkMapForKeyAndType(key, ConfigStoredType::Double);
+          },
+          R"(Returns true if specified key references a float value in this configuration.)")
+      .def(
+          "has_quat",
+          [](Configuration& self, const std::string& key) {
+            return self.checkMapForKeyAndType(key,
+                                              ConfigStoredType::MagnumQuat);
+          },
+          R"(Returns true if specified key references a Magnum::Quaternion value in this configuration.)")
+      .def(
+          "has_vec3",
+          [](Configuration& self, const std::string& key) {
+            return self.checkMapForKeyAndType(key,
+                                              ConfigStoredType::MagnumVec3);
+          },
+          R"(Returns true if specified key references a Magnum::Vector3 value in this configuration.)")
+      .def(
+          "has_rad",
+          [](Configuration& self, const std::string& key) {
+            return self.checkMapForKeyAndType(key, ConfigStoredType::MagnumRad);
+          },
+          R"(Returns true if specified key references a Magnum::Rad value in this configuration.)")
+      .def(
+          "has_subconfig", &Configuration::hasSubconfig,
+          R"(Returns true if specified key references an existing subconfiguration within this configuration.)")
+
+      .def(
+          "remove_bool", &Configuration::removeBool,
+          R"(Removes and returns boolean value corresponding to passed key, if found. Gives warning otherwise.)")
+      .def(
+          "remove_string", &Configuration::removeString,
+          R"(Removes and returns string value corresponding to passed key, if found. Gives warning otherwise.)")
+      .def(
+          "remove_int", &Configuration::removeInt,
+          R"(Removes and returns integer value corresponding to passed key, if found. Gives warning otherwise.)")
+      .def(
+          "remove_float", &Configuration::removeDouble,
+          R"(Removes and returns float value corresponding to passed key, if found. Gives warning otherwise.)")
+      .def(
+          "remove_vec3", &Configuration::removeVec3,
+          R"(Removes and returns Magnum::Vector3 value corresponding to passed key, if found. Gives warning otherwise.)")
+      .def(
+          "remove_quat", &Configuration::removeQuat,
+          R"(Removes and returns Magnum::Quaternion value corresponding to passed key, if found. Gives warning otherwise.)")
+      .def(
+          "remove_rad", &Configuration::removeRad,
+          R"(Removes and returns Magnum::Rad value corresponding to passed key, if found. Gives warning otherwise.)")
+      .def(
+          "remove_subconfig", &Configuration::removeSubconfig,
+          R"(Removes and returns subconfiguration corresponding to passed key, if found. Gives warning otherwise.)");
```

Comment:
I think these overloads aren't needed either anymore now that you have a single map for all -- just `foo.remove('bar')`. Or, in a more Pythonic way again, implement `__delitem__` / `__delattr__`:

```py
del foo['bar']
# or
del foo.bar
```

Reply:
I did have them separate so that I could "remove and return" but with your suggestion of py::cast above, I can now do this in a single method.